url458e40ff41_mmacfadden_java_baldwin_ga_tgz-pJ8-edu_sdsu_cs652_si_baldwin_SimulationEngineJ8/src/edu/sdsu/cs652/si/baldwin/MyFixedBinaryGene.java

```
@@ -0,0 +1,60 @@
+package edu.sdsu.cs652.si.baldwin;
+
+import org.jgap.Configuration;
+import org.jgap.Gene;
+import org.jgap.InvalidConfigurationException;
+import org.jgap.impl.FixedBinaryGene;
+
+public class MyFixedBinaryGene extends FixedBinaryGene {
+
+	private static final long serialVersionUID = 1L;
+
+	public MyFixedBinaryGene(Configuration a_config, int a_length)
+			throws InvalidConfigurationException {
+		super(a_config, a_length);
+	}
+
+
+	public MyFixedBinaryGene(Configuration a_config, FixedBinaryGene a_toCopy)
+			throws InvalidConfigurationException {
+		super(a_config, a_toCopy);
+	}
+
+
+	public int size() {
+		return getLength();
+	}
+
+	public Object clone() {
+		try {
+			return new MyFixedBinaryGene(getConfiguration(), this);
+		} catch (InvalidConfigurationException iex) {
+			throw new IllegalStateException(iex.getMessage());
+		}
+	}
+	
+	protected Gene newGeneInternal() {
+	    try {
+	    	MyFixedBinaryGene result = new MyFixedBinaryGene(getConfiguration(), getLength());
+	      return result;
+	    }
+	    catch (InvalidConfigurationException iex) {
+	      throw new IllegalStateException(iex.getMessage());
+	    }
+	  }
+	
+	public MyFixedBinaryGene substring(final int a_from, final int a_to) {
+	    try {
+	      int len = checkSubLength(a_from, a_to);
+	      MyFixedBinaryGene substring = new MyFixedBinaryGene(getConfiguration(), len);
+	      for (int i = a_from; i <= a_to; i++) {
+	        substring.setUnchecked(i - a_from, getUnchecked(i));
+	      }
+	      return substring;
+	    }
+	    catch (InvalidConfigurationException iex) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)